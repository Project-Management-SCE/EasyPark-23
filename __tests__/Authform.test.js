import React from 'react';
import { create } from 'react-test-renderer';
import AuthForm from '../components/Auth/AuthForm';
import Button from '../components/ui/Button';
import Input from '../components/Auth/Input';
import { View } from 'react-native';

describe('AuthForm', () => {
  it('displays the correct button text based on isLogin prop', () => {
    const submitHandler = jest.fn();
    let isLogin = true;
    const component = create(
      <AuthForm
        isLogin={isLogin}
        onSubmit={submitHandler}
        credentialsInvalid={{}}
      />
    );
    const instance = component.root;

    const button = instance.findByType(Button);
    expect(button.props.children).toEqual('התחברות');

    isLogin = false;
    component.update(
      <AuthForm
        isLogin={isLogin}
        onSubmit={submitHandler}
        credentialsInvalid={{}}
      />
    );

    expect(button.props.children).toEqual('הרשמה');
  });
});

// describe("Input Validation" , () => {
//   const mockSubmitHandler = jest.fn();

  // describe("Valid Input" , () => {
  //   const component = create(
  //     <View>
  //       <Input
  //         label="Email Address"
  //         onUpdateValue={() => {}}
  //         value="test@example.com"
  //         keyboardType="email-address"
  //         isInvalid={false}
  //       />
  //       <Input
  //         label="Password"
  //         onUpdateValue={() => {}}
  //         secure
  //         value="password123"
  //         isInvalid={false}
  //       />
  //     </View>,
  //   );
  //   it("render without errors" , () => {
  //     expect(component.toJSON()).toMatchSnapshot();
  //   });

  //   it('calls submit handler on button press', () => {
  //     const button = component.root.findByProps({ testID: 'loginBtn' });
  //     button.props.onPress();
  //     expect(mockSubmitHandler).toHaveBeenCalled();
  //   });

  // })
// })
