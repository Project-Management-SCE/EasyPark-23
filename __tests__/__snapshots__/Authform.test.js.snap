// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Validation Valid Input render without errors 1`] = `
<View>
  <View
    style={
      {
        "marginVertical": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "marginBottom": 4,
            "textAlign": "right",
          },
          false,
        ]
      }
    >
      Email Address
    </Text>
    <TextInput
      autoCapitalize="none"
      keyboardType="email-address"
      onChangeText={[Function]}
      style={
        [
          {
            "backgroundColor": "#f0f0f0",
            "borderRadius": 4,
            "fontSize": 16,
            "paddingHorizontal": 6,
            "paddingVertical": 8,
          },
          false,
        ]
      }
      value="test@example.com"
    />
  </View>
  <View
    style={
      {
        "marginVertical": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "marginBottom": 4,
            "textAlign": "right",
          },
          false,
        ]
      }
    >
      Password
    </Text>
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        [
          {
            "backgroundColor": "#f0f0f0",
            "borderRadius": 4,
            "fontSize": 16,
            "paddingHorizontal": 6,
            "paddingVertical": 8,
          },
          false,
        ]
      }
      value="password123"
    />
  </View>
</View>
`;
